File: java/src/com/company/context/ContextFactory.java
package com.company.context;

import java.lang.reflect.InvocationTargetException;

public class ContextFactory implements IContextFactory {
    private final IContextBuilder mBuilder;

    ContextFactory(IContextBuilder builder) {
        mBuilder = builder;
    }

    public IContext create(Object config) throws IllegalBeanException, IllegalConfigException, IllegalAccessException,
            IllegalArgumentException, InvocationTargetException {
        return mBuilder.build(config);
    }
}

File: dotNet/Lib/ContextFactory.cs
namespace Com.Company.Context {
    public class ContextFactory : IContextFactory {
        private readonly IContextBuilder mBuilder;

        public ContextFactory(IContextBuilder builder) {
            mBuilder = builder;
        }

        public IContext Create(object config) {
            return mBuilder.Build(config);
        }
    }
}